### PR TITLE
feat: sort results

### PR DESCRIPTION
**Pull request by jrachelr**
## PR Checklist

- [x] Addresses an existing open issue: fixes #000
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Adds sort feature to search results table.

I think everything inside the `<th>` should be extracted into it's own component, but before doing that I'd like to get some feedback on the UI.

### CODE REVIEW

File: src/components/ResultDisplay.module.css [hunk 1]
```diff
@@ -28,3 +30,20 @@
 	padding: 1rem;
 	overflow: auto;
 }
+
+.sortWidget {
+	background-color: transparent;
+	border-width: 0;
+	padding-left: 0.5em;
+	padding-right: 0.5em;
+	transition: transform 0.05s ease-out;
```
Comment (JoshuaKGoldberg) [inline src/components/ResultDisplay.module.css]: > `0.05s`

So fast! Maybe, a fifth or fourth of a second?

File: src/components/ResultDisplay.module.css [hunk 2]
```diff
@@ -28,3 +30,20 @@
 	padding: 1rem;
 	overflow: auto;
 }
+
+.sortWidget {
```
Comment (JoshuaKGoldberg) [inline src/components/ResultDisplay.module.css]: Request: `cursor: pointer`? Buttons don't have it by default.

File: src/components/ResultDisplay.tsx [hunk 3]
```diff
@@ -41,22 +56,82 @@ export function ResultDisplay({ result }: ResultDisplayProps) {
 			<table className={styles.estimates}>
 				<thead>
 					<tr>
-						<th className={styles.th}>Package Name</th>
-						<th className={styles.th}>Estimate</th>
-						<th className={styles.th}>Status</th>
+						<th className={styles.th}>
+							Package Name
```
Comment (JoshuaKGoldberg) [inline src/components/ResultDisplay.tsx]: Can the text go inside the button too? I'm finding it surprising that clicking _Package Name_ doesn't trigger anything.

**Review by JoshuaKGoldberg — COMMENTED** [review]
Looks good so far! Just a few requests.

File: src/components/ResultDisplay.module.css (hunk 1 above)
Comment (jrachelr) [reply src/components/ResultDisplay.module.css]: Ok! Updated to .25s

File: src/components/ResultDisplay.module.css (hunk 2 above)
Comment (jrachelr) [reply src/components/ResultDisplay.module.css]: ✅done!

File: src/components/ResultDisplay.tsx (hunk 3 above)
Comment (jrachelr) [reply src/components/ResultDisplay.tsx]: Done! This was a little tricky, lots of unexpected side effects in the CSS. Let me know what you think! I am unsure about the bold styling for `isActive`, I think it's helpful to let the user know which column is selected. I played with a few options and this is what I thought looked best.

**Pull request merged**